manager/manager.go

```diff
@@ -483,6 +514,107 @@ func (m *Manager) Stop(ctx context.Context) {
 	// mutex is released and Run can return now
 }
 
+func (m *Manager) joinAndStart(ctx context.Context) error {
+	return m.config.SecurityConfig.KeyWriter().UpdateHeaders(func(headers map[string]string, kek []byte) error {
+		if err := m.keyRotator.LoadFromHeaders(headers, kek); err != nil {
+			return err
+		}
+
+		// if there is no DEK, create a new one
+		dekStr, foundDEK := headers[defaultRaftDEKKey]
+		if !foundDEK {
+			newDEK, err := encryption.GenerateSecretKey()
+			if err != nil {
+				return err
+			}
+			if headers[defaultRaftDEKKey], err = encodePEMHeaderValue(newDEK, kek); err != nil {
+				return err
+			}
+			return m.raftNode.JoinAndStart(ctx, newDEK)
+		}
+
+		if pendingKey := m.keyRotator.GetNewKey(); pendingKey != nil {
+			// if there is a pending header, try decrypting with that first - if that doesn't work, fall back
+			// on the normal dek but finish rotating
+			err := m.raftNode.JoinAndStart(ctx, pendingKey)
+			switch err.(type) {
+			case encryption.ErrCannotDecode:
+				// if the dek hasn't been used to snapshot, fall back to the original dek - a rotation
+				// still needs to happen
+				break
+			case nil:
+				// we've successfully joined - that means the previous snapshot had succeeded - finish
+				// the rotation
+				return m.keyRotator.FinishKeyRotation(pendingKey)
+			default:
+				// we failed to join but not due to a decoding error
+				return err
+			}
+		}
+
+		// there is a regular header - use that to decrypt and start
+		dek, err := decodePEMHeaderValue(dekStr, kek)
+		if err != nil {
+			return err
+		}
+		return m.raftNode.JoinAndStart(ctx, dek)
+	})
+}
+
+func (m *Manager) watchForKEKChanges(ctx context.Context) {
+	clusterWatch, clusterWatchCancel := state.Watch(m.raftNode.MemoryStore().WatchQueue(),
```

Comment:
I **think** you're supposed to `GetCluster` first then watch it by its ID - @aaronlehmann ?